pkg/controller/certificaterequests/acme/acme.go

```diff
@@ -161,12 +161,9 @@ func (a *ACME) Sign(ctx context.Context, cr *v1alpha2.CertificateRequest, issuer
 
 	// If the acme order has failed then so too does the CertificateRequest meet the same fate.
 	if acme.IsFailureState(order.Status.State) {
-		message := fmt.Sprintf("Failed to wait for order resource %s/%s to become ready",
-			expectedOrder.Namespace, expectedOrder.Name)
-		err := fmt.Errorf("order is in %q state", order.Status.State)
-
+		message := fmt.Sprintf("Failed to wait for order resource %q to become ready", expectedOrder.Name)
+		err := fmt.Errorf("order is in %q state: %s", order.Status.State, order.Status.Reason)
 		a.reporter.Failed(cr, err, "OrderFailed", message)
```

Comment:
In the PR description you say that this extra event information will also show up in `kubectl describe certificate`, but I only the error message being associated with a `cr` resource...a `CertificateRequest` Or does the reporter also find the associated Certificate and Order resources and add events for those too?

Reply:
It is surfaced as the Certificate controller extracts the reason for a failed CertificateRequest and stores it on the *Certificates* status as well 😄  (so no, the reporter does not log against the Order and Certificate, their own respective controllers handle it)